multi_map_it_test.go

```diff
@@ -539,6 +539,49 @@ func TestMultiMap_ContainsEntry(t *testing.T) {
 	})
 }
 
+func TestMultiMap_ValueCount(t *testing.T) {
+	it.MultiMapTester(t, func(t *testing.T, m *hz.MultiMap) {
+		ctx := context.Background()
+		targetValues := []interface{}{"v1", "v2", "v3"}
+		targets := []types.Entry{
+			types.NewEntry("key1", "v1"),
+			types.NewEntry("key1", "v2"),
+			types.NewEntry("key1", "v3"),
+		}
+		for _, entry := range targets {
+			assert.True(t, it.MustBool(m.Put(ctx, entry.Key, entry.Value)), "multi-map put failed")
+		}
+		values := it.MustSlice(m.Get(ctx, "key1"))
+		assert.ElementsMatch(t, targetValues, values)
```

Comment:
We don't need to get and assert values for "key1" since this is the test for "valueCount". They are tested in separate tests.
How about asserting against a constant like `assert.Equal(t, 3, count)` ?

Reply:
In this way, I tried to ensure that `ValueCount` does not alter the actual content of the multi-map. 

- Lets suppose I have a multi-map with given entry below:
`key -> 1, 2, 3`
- Then, `ValueCount` method change the last element in some weird way.
`key -> 1, 2, 4`
- In this situation, if I don't check whether content of the multi-map changes, I could not notice that buggy case

Reply:
I think this somewhat feels out of scope since it tests whether ValueCount is implemented correctly on server-side.
I would appreciate one more reviewer idea on this 🙏

Reply:
Based on our discussion, I'm convinced that excluding those cases make more sense I guess. Actual implementation is  short and checking whether we use correct codec or not by reviews not in the test actually. I can even use wrong codec and alter another data structure. My assertion is making mistake on implementation in the context of given key has higher probability than other mistake case space. 

I wonder other reviewers ideas on this as well. @yuce @srknzl 

prev imp: [7818ab7](https://github.com/hazelcast/hazelcast-go-client/pull/734/commits/7818ab7e3542bd330b0dabcf368dc648c5321ec1)
cur imp: [9ab0e17](https://github.com/hazelcast/hazelcast-go-client/pull/734/commits/9ab0e1786fcc3f4474facea567467d9e521cc78b)

Reply:
I agree with @utku-caglayan on this one. No need to check whether `ValueCount` modifies the map.